src/parser/mumexmlparser.cpp

```diff
@@ -172,23 +176,36 @@ void MumeXmlParser::parse(const TelnetData &data, const bool isGoAhead)
         m_tempCharacters.clear();
     }
     if (!m_lineToUser.isEmpty()) {
+        // Simplify the output and run actions
+        QByteArray temp = m_lineToUser;
+        if (!getConfig().parser.removeXmlTags) {
+            stripXmlEntities(temp);
+        }
+        QString tempStr = temp;
+        tempStr = normalizeStringCopy(tempStr.trimmed());
+        if (m_snoopChar.has_value() && tempStr.length() > 3 && tempStr.at(0) == '&'
+            && tempStr.at(1) == m_snoopChar.value() && tempStr.at(2) == ' ') {
+            // Remove snoop prefix (i.e. "&J Exits: north.")
+            tempStr = tempStr.mid(3);
+        }
+
         sendToUser(m_lineToUser, isGoAhead);
+        printTimersExtensions(tempStr, isGoAhead);
+        parseMudCommands(tempStr);
+    }
+}
 
-        {
-            // Simplify the output and run actions
-            QByteArray temp = m_lineToUser;
-            if (!getConfig().parser.removeXmlTags) {
-                stripXmlEntities(temp);
-            }
-            QString tempStr = temp;
-            tempStr = normalizeStringCopy(tempStr.trimmed());
-            if (m_snoopChar.has_value() && tempStr.length() > 3 && tempStr.at(0) == '&'
-                && tempStr.at(1) == m_snoopChar.value() && tempStr.at(2) == ' ') {
-                // Remove snoop prefix (i.e. "&J Exits: north.")
-                tempStr = tempStr.mid(3);
-            }
-            parseMudCommands(tempStr);
-        }
+void MumeXmlParser::printTimersExtensions(const QString currentString, bool isGoAhead)
+{
+    // lvl100 hungry/thirsty?
+    // shall this be simplified somehow?
+    static QRegExp statExp("Needed: * Alert: *.", Qt::CaseSensitive, QRegExp::Wildcard);
+    static QRegExp statExpCond("Needed: * Alert: *. Cond*", Qt::CaseSensitive, QRegExp::Wildcard);
+    static QRegExp lvl100statExp("Gold: *. Alert: *.", Qt::CaseSensitive, QRegExp::Wildcard);
+
+    if (statExp.exactMatch(currentString) || lvl100statExp.exactMatch(currentString)
+        || statExpCond.exactMatch(currentString)) {
+        sendToUser(m_timers.getStatCommandEntry(), isGoAhead);
```

Comment:
isGoAhead is only for prompts so we don't need to pass this argument through. Given this, we can also keep this logic confined to parseMudCommands()

Reply:
Not yet sure I understood the hint about keeping the logic in parseMudCommands (need to take a look at the code). But I guess i can remove the isGoAhead. This is for sure only about injecting additional lines right after the matching one.

Reply:
Ah, let me give the backstory. The general Parser file is getting huge so I'm trying to isolate similar logic to other files